### PR TITLE
EVA-3225 - Add native execution for validation and refactor

### PR DESCRIPTION
**Pull request by apriltuesday**

### CODE REVIEW

File: eva_sub_cli/validators/native_validator.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def __init__(self, mapping_file, output_dir, metadata_json=None, metadata_xlsx=None,
+                 vcf_validator_path='vcf_validator', assembly_checker_path='vcf_assembly_checker',
+                 biovalidator_path='biovalidator', submission_config=None):
+        super().__init__(mapping_file, output_dir, metadata_json=metadata_json, metadata_xlsx=metadata_xlsx,
+                         submission_config=submission_config)
+        self.vcf_validator_path = vcf_validator_path
+        self.assembly_checker_path = assembly_checker_path
+        self.biovalidator_path = biovalidator_path
+
+    def validate(self):
```
[the remark this answers is not in the document]
Comment (apriltuesday) [reply eva_sub_cli/validators/native_validator.py]: yes indeed... thanks!

File: eva_sub_cli/validators/validator.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import csv
+import os
+
+from eva_sub_cli.reporter import Reporter
+
+
+VALIDATION_OUTPUT_DIR = "validation_output"
+
+
+class Validator(Reporter):
```
[the remark this answers is not in the document]
Comment (apriltuesday) [reply eva_sub_cli/validators/validator.py]: True, I've combined them into just `Validator`

File: eva_sub_cli/main.py [hunk 3]
```diff
@@ -52,11 +55,17 @@ def orchestrate_process(submission_dir, vcf_files_mapping, vcf_files, assembly_f
...
-        with DockerValidator(vcf_files_mapping, submission_dir, metadata_json, metadata_xlsx,
-                             submission_config=sub_config) as validator:
-            validator.validate()
-            validator.create_reports()
-            validator.update_config_with_validation_result()
+        if executor == DOCKER:
+            with DockerValidator(vcf_files_mapping, submission_dir, metadata_json, metadata_xlsx,
+                                 submission_config=sub_config) as validator:
+                validator.validate()
+                validator.report()
```
[the remark this answers is not in the document]
Comment (apriltuesday) [reply eva_sub_cli/main.py]: Added to the combined `Validator` - I'm not sure we still need plain `validate` and `report`, I left them in for now...

File: eva_sub_cli/validators/native_validator.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        self.run_validator()
+
+    def run_validator(self):
+        self.verify_files_present()
```
[the remark this answers is not in the document]
Comment (apriltuesday) [reply eva_sub_cli/validators/native_validator.py]: done

File: eva_sub_cli/validators/native_validator.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+from eva_sub_cli.validators.validator import Validator
+
+logger = logging_config.get_logger(__name__)
+
+
+class NativeValidator(Validator):
+
+    def __init__(self, mapping_file, output_dir, metadata_json=None, metadata_xlsx=None,
+                 vcf_validator_path='vcf_validator', assembly_checker_path='vcf_assembly_checker',
+                 biovalidator_path='biovalidator', submission_config=None):
```
Comment (tcezard) [inline eva_sub_cli/validators/native_validator.py:16]: Do you anticipate that the executable paths could be set by the parent maybe from command line args ?

**Review by tcezard — APPROVED** [review]
Great work!

**Review by nitin-ebi — APPROVED** [review]

**Pull request merged**